### PR TITLE
Get rid of unused get_user route and favor Result combinators over ma…

### PR DESCRIPTION
**Pull request by cedric-h**
…tches

### CODE REVIEW

**Review by Muirrum — CHANGES_REQUESTED** [review]

File: src/main.rs [hunk 1]
```diff
@@ -1,22 +1,47 @@
...
+
+fn to_doc<S: serde::Serialize>(s: &S) -> Result<bson::Document, RequestError> {
+    match bson::to_bson(s)? {
+        bson::Bson::Document(d) => Ok(d),
+        not_doc => Err(RequestError::NotDocument(not_doc)),
+    }
+}
+
+#[actix_rt::test]
+async fn test_get_user() -> Result<(), ServiceError> {
```
[the remark this answers is not in the document]
Comment (cedric-h) [reply src/main.rs]: done

File: Cargo.toml [hunk 2]
```diff
@@ -7,10 +7,13 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-fll-scoring = "0.1.0"
 actix-web = "2.0"
 actix-rt = "1.0"
+hcor = { path = "../hcor/" , features = [ "actix", "mongo" ] }
```
[the remark this answers is not in the document]
Comment (cedric-h) [reply Cargo.toml]: done

File: src/hackstead/mod.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+use actix_web::{get, web, HttpResponse};
+use hcor::{ServiceError, Hackstead, UserContact};
+
+#[get("/hackstead/")]
+pub async fn get_hackstead(form: web::Json<UserContact>) -> Result<HttpResponse, ServiceError> {
+    log::debug!("servicing get_hackstead request");
+
+    let slack_id = form.slack().ok_or(ServiceError::bad_request("no slack"))?;
```
Comment (Muirrum) [inline src/hackstead/mod.rs:8]: this should work even without slack users.

**Review by Muirrum — CHANGES_REQUESTED** [review]

**Comment by Muirrum** [comment]
run cargo fmt as well

File: src/hackstead/mod.rs (hunk 3 above)
Comment (cedric-h) [reply src/hackstead/mod.rs:8]: yes but prototyping

File: Cargo.toml [hunk 4]
```diff
@@ -6,11 +6,31 @@ edition = "2018"
...
+path = "./src/from_csv.rs"
+name = "from_csv"
+
 [dependencies]
-fll-scoring = "0.1.0"
 actix-web = "2.0"
 actix-rt = "1.0"
 bson = "1.0"
-serde = { version = "1.0", features = ["derive"]}
+serde = { version = "1.0.114", features = ["derive"] }
```
Comment (Muirrum) [inline Cargo.toml:17]: Any reason you got more specific?

**Comment by Muirrum** [comment]
Run `cargo fmt`

File: Cargo.toml (hunk 4 above)
Comment (cedric-h) [reply Cargo.toml:17]: no this was a stupid thing to do

Comment (cedric-h) [reply Cargo.toml:17]: great catch!

File: .build.yml [hunk 5]
```diff
@@ -4,15 +4,15 @@ packages:
...
-    rustup run stable cargo fetch
-- stable: |
-    rustup default stable
-    cd hackagotchi-backend/
+    rustup toolchain install nightly
+    cd backend/
+    rustup run nightly cargo fetch
+- nightly: |
+    rustup default nightly
+    cd backend/
```
Comment (Muirrum) [inline .build.yml:15]: this has been needed for a while

**Review by Muirrum — APPROVED** [review]

**Pull request merged**